src/test/cypress/e2e/exercises/ExerciseImport.cy.ts

```diff
@@ -0,0 +1,180 @@
+            });
+            courseManagementRequest.createQuizExercise({ course }, [multipleChoiceQuizTemplate]).then((response) => {
+                quizExercise = response.body;
+            });
+            courseManagementRequest.createModelingExercise({ course }).then((response) => {
+                modelingExercise = response.body;
+            });
+            courseManagementRequest.createProgrammingExercise({ course }).then((response) => {
+                programmingExercise = response.body;
+            });
+            courseManagementRequest.createCourse(true).then((response) => {
+                secondCourse = convertCourseAfterMultiPart(response);
+                courseManagementRequest.addStudentToCourse(secondCourse, studentOne);
+                courseManagementRequest.addInstructorToCourse(secondCourse, instructor);
+            });
+        });
+    });
+
+    it('Imports text exercise', () => {
+        cy.login(instructor, `/course-management/${secondCourse.id}/exercises`);
+        courseManagementExercises.importTextExercise();
+        courseManagementExercises.clickImportExercise(textExercise.id!);
+
+        checkField('#field_title', textExercise.title!);
+        checkField('#field_points', textExercise.maxPoints!);
+
+        textExerciseCreation.setReleaseDate(dayjs());
+        textExerciseCreation.setDueDate(dayjs().add(1, 'days'));
+        textExerciseCreation.setAssessmentDueDate(dayjs().add(2, 'days'));
+
+        textExerciseCreation.import().then((request: Interception) => {
+            const exercise = request.response!.body;
+            cy.login(studentOne, `/courses/${secondCourse.id}`);
+            courseOverview.startExercise(exercise.id!);
+            courseOverview.openRunningExercise(exercise.id!);
+            cy.fixture('loremIpsum.txt').then((submission) => {
+                textExerciseEditor.shouldShowNumberOfWords(0);
+                textExerciseEditor.shouldShowNumberOfCharacters(0);
+                textExerciseEditor.typeSubmission(exercise.id!, submission);
+                textExerciseEditor.shouldShowNumberOfWords(100);
+                textExerciseEditor.shouldShowNumberOfCharacters(591);
+                textExerciseEditor.submit().then((request: Interception) => {
+                    expect(request.response!.body.text).to.eq(submission);
+                    expect(request.response!.body.submitted).to.be.true;
+                    expect(request.response!.statusCode).to.eq(200);
+                });
+            });
+        });
+    });
+
+    it('Imports quiz exercise', () => {
+        cy.login(instructor, `/course-management/${secondCourse.id}/exercises`);
+        courseManagementExercises.importQuizExercise();
+        courseManagementExercises.clickImportExercise(quizExercise.id!);
+
+        checkField('#quiz-title', quizExercise.title!);
+        checkField('#quiz-duration-minutes', quizExercise.duration! / 60);
+
+        cy.wait(500);
```

Comment:
Can we replace that with a reloadUntil..., or is simply waiting besser here?

Reply:
Usually yes, but we don't want to reload the page here. The wait is necessary since the date field needs some time to initialize. Usually the date is set later in the process, but since it is the only field we change here, this wait is needed.